### PR TITLE
deepin desktop environment: Minor update

### PR DESCRIPTION
**Pull request by wineee**
## Description of changes

https://www.deepin.org/en/deepin-v23-beta3-released/

image
image

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by wineee** [comment]
@ofborg test deepin

**Comment by wineee** [comment]
@ofborg test deepin

**Comment by wineee** [comment]
@ofborg test deepin

File: pkgs/desktops/deepin/go-package/dde-api/default.nix [hunk 1]
```diff
@@ -21,24 +21,16 @@
...
   src = fetchFromGitHub {
     owner = "linuxdeepin";
     repo = pname;
     rev = version;
-    hash = "sha256-kdf1CoZUyda6bOTW0WJTgaXYhocrjRU9ptj7i+k8aaQ=";
+    hash = "sha256-ht5IaXi4nz0/U1zqp4JTiDkQ3NB69q24MgWfu45SpoY=";
   };
 
-  patches = [
-    (fetchpatch {
```
Comment (SuperSandro2000) [inline pkgs/desktops/deepin/go-package/dde-api/default.nix:34]: Please remove the unused fetchpatch from inputs.

File: pkgs/desktops/deepin/default.nix [hunk 2]
```diff
@@ -1,4 +1,4 @@
-{ lib, pkgs, config, libsForQt5 }:
+{ lib, pkgs, config, libsForQt5, qt6Packages }:
```
Comment (SuperSandro2000) [inline pkgs/desktops/deepin/default.nix:1]: Are not all upgraded to qt6?

File: pkgs/desktops/deepin/apps/deepin-screen-recorder/default.nix [hunk 3]
```diff
@@ -78,6 +78,10 @@ stdenv.mkDerivation rec {
     gst-plugins-good
   ]);
 
+
```
Comment (SuperSandro2000) [inline pkgs/desktops/deepin/apps/deepin-screen-recorder/default.nix:81]:
```suggestion
```

File: pkgs/desktops/deepin/apps/deepin-compressor/default.nix [hunk 4]
```diff
@@ -20,20 +20,24 @@
...
   src = fetchFromGitHub {
     owner = "linuxdeepin";
     repo = pname;
     rev = version;
-    hash = "sha256-8qfpNM2rci4subdodxfJZLP3OvAxXl7QRl4MHGr15nA=";
+    hash = "sha256-XNhG28VZifQrl3TZfx/OHnsAOo0eKrhGKDk+OjOYD8k=";
   };
 
+  patches = [
+     ./0001-fix-build-on-new-dtk.diff
```
Comment (SuperSandro2000) [inline pkgs/desktops/deepin/apps/deepin-compressor/default.nix:33]:
```suggestion
    ./0001-fix-build-on-new-dtk.diff
```

**Review by SuperSandro2000 — APPROVED** [review]

**Comment by wineee** [comment]
@ofborg test deepin

File: pkgs/desktops/deepin/default.nix (hunk 2 above)
Comment (wineee) [reply pkgs/desktops/deepin/default.nix:1]: Only `dde-application-manager` use qt6 so far

There are currently some new projects that only support qt6, such as `dde-shell`, `treeland`, they will take the place of `dde-dock`, `deepin-kwin`. But still under early development

The latest version of dde-launchpad supports qt 6, but it need qt >= 6.7 for [this patch](https://codereview.qt-project.org/c/qt/qtdeclarative/+/414503), so I still use qt5 to build launchpad

**Comment by wineee** [comment]
@ofborg test deepin

**Pull request merged**